### PR TITLE
Add redcap_bridge (replacement of #8)

### PR DESCRIPTION
**Pull request by JuliaSprenger**

### CODE REVIEW

File: redcap_bridge/project_building.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    return output
+
+
+def customize_project(project_built_csv, customization_csv, output_file=None):
+    """
+    Fill in a built project csv with project specific customizations.
+
+    This can be used to e.g. change the default values of fields or customize
+    the list of experimenters to be selected
+
```
Comment (Slowblitz) [inline redcap_bridge/project_building.py:75]: I think there is some kind of wrong indentation no  ?

File: redcap_bridge/project_validation.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    Parameters
+    ----------
+    record_csv: path
+        path to the record csv of that instrument
+    template_csv: path
+        path to the template csv of an instrument
+
+    Returns
+    -------
+    True
```
Comment (Slowblitz) [inline redcap_bridge/project_validation.py:80]: This function always returns true  ?

File: redcap_bridge/test_redcap/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    Parameters
+    ----------
+    clean: (bool)
+        Remove test folder first in case it exists.
+
+    Returns
+    -------
+    test_directory: (str)
+        path of the test directory
```
Comment (Slowblitz) [inline redcap_bridge/test_redcap/test_utils.py:24]:
```suggestion
   str
       path of the test directory
```

File: redcap_bridge/test_redcap/test_server_interface.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+from redcap_bridge.server_interface import upload_datadict, download_records
+from redcap_bridge.test_redcap.test_utils import (test_directory,
+                                                  initialize_test_dir)
+from redcap_bridge.utils import map_header_csv_to_json
+
+SERVER_CONFIG_YAML = (
+        test_directory / 'testfiles' / 'TestProject' / 'project.json').resolve()
+
+
+@pytest.fixture
```
Comment (Slowblitz) [inline redcap_bridge/test_redcap/test_server_interface.py:16]: I'm not sure if I understand the fixture thing

File: redcap_bridge/server_interface.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                         f'and "json".')
+
+
+
+
+
+
+
+
+
```
Comment (Slowblitz) [inline redcap_bridge/server_interface.py:89]:
```suggestion


```

File: redcap_bridge/test_redcap/test_utils.py (hunk 3 above)
Comment (JuliaSprenger) [reply redcap_bridge/test_redcap/test_utils.py:24]: No, the standard should be 1 tab = 4 spaces, what were you comparing to? maybe the other code would need to have the indentation fixed..

Comment (Slowblitz) [reply redcap_bridge/test_redcap/test_utils.py:24]: Yes sorry it's bit hard in the web interface =/ I've updated the code

File: redcap_bridge/test_redcap/test_server_interface.py (hunk 4 above)
Comment (JuliaSprenger) [reply redcap_bridge/test_redcap/test_server_interface.py:16]: Fixtures are a pytest concept to create specific states for tests. I am not an expert myself, but I guess it is similar to the unittest setUp etc...

**Comment by JuliaSprenger** [comment]
\o/ test are passing!

**Comment by Slowblitz** [comment]
Yeaha ! 
\o/

**Comment by SylvainTakerkart** [comment]
I'm happy also, but you're gonna have to explain to me what's redcap_bridge... Ahah ;)

File: redcap_bridge/project_building.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+   customization_csv: str
+       The filepath to the csv containing the project customizations
+   output_file: str
+       The path to save the combined csv. Default: None
+
+   Returns
+   -------
+   dataframe 
+       pandas dataframe csv representation of the customized project definition
+    """
```
Comment (JuliaSprenger) [inline redcap_bridge/project_building.py:89]:
```suggestion
    Parameters
    ----------
    project_built_csv: str
        The filepath to the csv containing the built project (see also `build_project`)
    customization_csv: str
        The filepath to the csv containing the project customizations
    output_file: str
        The path to save the combined csv. Default: None

    Returns
    -------
    dataframe
        pandas dataframe csv representation of the customized project definition
    """
```

File: redcap_bridge/project_building.py (hunk 1 above)
Comment (JuliaSprenger) [reply redcap_bridge/project_building.py:75]: Yes I think I fixed it now.

File: redcap_bridge/project_validation.py (hunk 2 above)
Comment (JuliaSprenger) [reply redcap_bridge/project_validation.py:80]: Currently, yes. We had the discussion about potential future boolean values already in the original PR #8 ...

**Pull request merged**